src/platforms/atomic-kms/server/kms/atomic_kms_output.cpp

```diff
@@ -647,6 +654,43 @@ std::vector<uint8_t> edid_for_connector(mir::Fd const& drm_fd, uint32_t connecto
 
     return edid;
 }
+
+auto formats_for_output(mir::Fd const& drm_fd, mgk::DRMModeConnectorUPtr const& connector) -> std::vector<mg::DRMFormat>
+{
+    auto [_, plane] = mgk::find_crtc_with_primary_plane(drm_fd, connector);
+
+    mgk::ObjectProperties plane_props{drm_fd, plane->plane_id, DRM_MODE_OBJECT_PLANE};
+
+    if (!plane_props.has_property("IN_FORMATS"))
+    {
+        return {mg::DRMFormat{DRM_FORMAT_ARGB8888}, mg::DRMFormat{DRM_FORMAT_XRGB8888} };
+    }
+
+    PropertyBlobData format_blob{drm_fd, static_cast<uint32_t>(plane_props["IN_FORMATS"])};
+    drmModeFormatModifierIterator iter;
+    memset(&iter, 0, sizeof(iter));
```

Comment:
Using `memset()` makes assumptions of about the representation of data members that are not guaranteed by the language.  value-initialization should do the job:

```suggestion
    drmModeFormatModifierIterator iter{};
```